src/test_pages/BasePage.ts

```diff
@@ -12,7 +12,7 @@ export abstract class BasePage {
   protected async clickOnElement(element: Locator, timeout?: number) {
     // TODO: make sure that element.toString() doesn't make [object Object]
     await test.step(`Click on ${element.toString()}`, async () => {
-      await element.click({ timeout: timeout || 0 })
+      await element.click({ timeout })
```

Comment:
Looks great!
I took a look at Playwright documentation noticed a small thing,  ```timeout``` will default to ```0``` when undefined, 
If we want to change this behavior we can also modify the ```playwright.config.ts``` -
 ``` { use:  { ..., actionTimeout: 10000 }}```